packages/client/src/components/ToolDetailsPage/ToolDetailsPage.component.js

```diff
@@ -0,0 +1,124 @@
+import React, { useState, useEffect } from 'react';
+import { useParams } from 'react-router-dom';
+import getApiBaseUrl from '../../utils/getApiBaseURL';
+import './ToolDetailsPage.css';
+
+export const ToolDetailsPage = () => {
+  const [tool, setTool] = useState({});
+  const [isLoading, setIsLoading] = useState(true);
+  const [isDataEmpty, setisDataEmpty] = useState(false);
+  const params = useParams();
+
+  useEffect(() => {
+    async function fetchTool(id) {
+      const response = await fetch(`${getApiBaseUrl()}/api/tools/${id}`);
+      const result = await response.json();
+      return result;
+    }
+    function setFetchTool() {
+      setIsLoading(true);
+
+      fetchTool(params.id)
+        .then((data) => {
+          if (typeof data === 'object') {
+            setisDataEmpty(false);
+            const toolWithCategries = data[0];
+            toolWithCategries.categories = data.categories;
+            setTool(toolWithCategries);
+          } else {
+            setisDataEmpty(true);
+          }
+        })
+        .then(() => {
+          setIsLoading(false);
+        });
+    }
+    setFetchTool();
+  }, [params.id]);
+
+  let pageContent;
+  if (isLoading) {
+    pageContent = <p className="message">Loading</p>;
+  } else if (isDataEmpty) {
+    pageContent = <p className="message">Tool not found</p>;
+  } else {
+    const toolCategoriesList = tool.categories
+      ? tool.categories.map((category, index) => (
+          <span key={category.name}>
+            {tool.categories.length - 1 === index
+              ? category.name
+              : `${category.name} // `}
+          </span>
+        ))
+      : '';
```

Comment:
If you expect the categories to ever be empty by any chance, would be good to give some indication to the UI, like "n/a".
Otherwise, this will render just `CATEGORIES:             `

Reply:
Fixed.